### PR TITLE
Add hcl/json syntax validation ability before executing terraform

### PR DESCRIPTION
**Pull request by zzxwill**
Check the syntax for hcl/json before creating a job to running
    terraform init && apply

### CODE REVIEW

**Review by FogDong — COMMENTED** [review]
Should these binary files be uploaded?

`controllers/configuration/terraform/darwin/terraform`
`controllers/configuration/terraform/linux/terraform `

**Comment by zzxwill** [comment]
> Should these binary files be uploaded?
> 
> `controllers/configuration/terraform/darwin/terraform`
> `controllers/configuration/terraform/linux/terraform `

Yes. In order to run `terraform validate` to check the syntax issue for HCL/JSON in Terraform pod, Linux one is needed. In order to help local development, darwin and windows ones are needed.

File: Makefile [hunk 1]
```diff
@@ -85,7 +85,7 @@ else
 CONTROLLER_GEN=$(shell which controller-gen)
 endif
 
-GOLANGCILINT_VERSION ?= v1.40.1
+GOLANGCILINT_VERSION ?= v1.42.0
```
Comment (zzxwill) [inline Makefile:88]: This changes is to fix linter issue `MkdirTemp not declared by package os`.

**Review by FogDong — COMMENTED** [review]
/lgtm

**Pull request merged**